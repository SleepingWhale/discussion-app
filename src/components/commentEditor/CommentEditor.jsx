import React, { Component, createRef } from 'react';
import PropTypes from 'prop-types';
import styles from './CommentEditor.module.css';
import { Button } from '../common';

export class CommentEditor extends Component {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      text: ''
    };
    this.textareaRef = createRef();
  }

  componentDidMount() {
    this.textareaRef.current.focus();
  }

  onChange = e => {
    this.setState({
      text: e.target.value
    });
  };

  onSubmit = () => {
    const { text } = this.state;
    const { onSubmit } = this.props;

    onSubmit(text);

    this.setState({
      text: ''
    });
  };

  render() {
    const { text } = this.state;
    const isSubmitDisabled = text.length === 0;

    return (
      <div className={styles.container}>
        <textarea
          className={styles.text}
          maxLength={1000}
          rows={3}
          placeholder="Type anything"
          onChange={this.onChange}
          value={text}
          ref={this.textareaRef}
        />
        <div className={styles.buttons}>
          <Button
            onClick={this.onSubmit}
            mode="primary"
            disabled={isSubmitDisabled}
          >
            <i className="material-icons md-18">send</i>
            <span className={styles.ml04}>send</span>
          </Button>
        </div>
      </div>
    );
  }
}
