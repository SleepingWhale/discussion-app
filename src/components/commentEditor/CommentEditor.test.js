import React from 'react';
import { mount } from 'enzyme';
import { CommentEditor } from './CommentEditor';

describe('CommentEditor', () => {
  it('should render correctly', () => {
    const onSubmitHandler = jest.fn();
    const component = mount(<CommentEditor onSubmit={onSubmitHandler} />);

    expect(component).toMatchSnapshot();
  });

  it('should not submit without content', () => {
    const onSubmitHandler = jest.fn();
    const component = mount(<CommentEditor onSubmit={onSubmitHandler} />);
    component.find('button').simulate('click');

    expect(onSubmitHandler).not.toHaveBeenCalled();
  });

  it('should submit with content', () => {
    const content = 'hello world';
    const onSubmitHandler = jest.fn();
    const component = mount(<CommentEditor onSubmit={onSubmitHandler} />);
    const input = component.find('textarea').at(0);

    input.instance().value = content;
    input.simulate('change');
    expect(component.state().text).toEqual(content);

    component.find('button').simulate('click');
    expect(onSubmitHandler).toHaveBeenCalledWith(content);
  });
});
