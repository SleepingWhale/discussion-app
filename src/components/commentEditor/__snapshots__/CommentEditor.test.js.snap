// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentEditor should render correctly 1`] = `
<CommentEditor
  onSubmit={[MockFunction]}
>
  <div
    className="container"
  >
    <textarea
      className="text"
      maxLength={1000}
      onChange={[Function]}
      placeholder="Type anything"
      rows={3}
      value=""
    />
    <div
      className="buttons"
    >
      <Button
        disabled={true}
        mode="primary"
        onClick={[Function]}
      >
        <button
          className="base primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <i
            className="material-icons md-18"
          >
            send
          </i>
          <span
            className="ml04"
          >
            send
          </span>
        </button>
      </Button>
    </div>
  </div>
</CommentEditor>
`;
